src/components/AppView/AppStatus.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from "react";
+
+import Check from "../../icons/Check";
+import Compass from "../../icons/Compass";
+import { IDeploymentStatus, IResource } from "../../shared/types";
+import "./AppStatus.css";
+
+interface IAppStatusProps {
+  deployments: Map<string, IResource>;
+}
+
+class AppStatus extends React.Component<IAppStatusProps> {
+  public render() {
+    let status = (
+      <span className="AppStatus AppStatus--success">
```

Comment:
I'd move the `AppStatus` elements and the check logic to other methods. We should keep render methods as simple as we can:

```
public isReady() {
  return (Object.keys(deployments).length > 0) {
    const allPodsReady = Object.keys(deployments).every(k => {
      const dStatus: IDeploymentStatus = deployments[k].status;
      return dStatus.availableReplicas === dStatus.replicas;
    });
}

public renderSuccessStatus() {
  return <span className="AppStatus AppStatus--success"> ... </span>
}

public renderDeployingStatus() {
  return <span className="AppStatus AppStatus--pending"> ... </span>
}

Reply:
:+1: